src/core-messages.lisp

```diff
@@ -158,7 +158,7 @@
     :documentation "Center time coordinate of the shape in seconds. Defaults to mid-point of pulse."
     :type :float
     :required t
-    :default nil)
+    :default 0.0)
```

Comment:
I suppose this [changes the API](https://github.com/rigetti/rpcq/blob/322cf061ec150203dcc1b93bc4667b6fbfef308c/src/rpcq.lisp#L215) - former instantiations without a value will now pass rather than fail, but not a big deal since this is a server. 

Regardless, for a required field, it seems more intuitive to signal an error before coercing to the default value  - then just drop the defaults from `defmessage` that don't have a reachable code path.

Reply:
I don't think the first point is true. The default here doesn't really count for anything, other than to satisfy the compiler's type-check. If you try to `make-instance` a message without a `:required t` slot you'll hit an error.

Your second point I agree with and I had tried that approach in a limited fashion, removing `:default` in this particular message. The problem is that if we drop the `:default` then the Python classes will potentially have non-default properties that follow default ones. For example, the `TemplateWaveform` message has some properties that are provided a default value (even though they're also required in the spec). Any message that inherits from `TemplateWaveform` is therefore /required/ to specify all properties with defaults, otherwise you hit an [error like this](https://bugs.python.org/issue39300). We could drop all `:default` from `:required t` messages (which logically seems sensible) but then we might mess with Python folks that were depending (erroneously) on those default values.

Of the options available, I think the proposed one is the smallest and the safest.

Reply:
That's an ugly corner of `dataclass`es that I didn't know about.